pandas/core/sparse/series.py

```diff
@@ -135,6 +135,29 @@ def __array_finalize__(self, obj):
         self.name = getattr(obj, 'name', None)
         self.fill_value = getattr(obj, 'fill_value', None)
 
+    # unary ops
+    # TODO: See if this can be shared
```

Comment:
@jreback @jbrockmendel do we have any kind of "dispatch unary ops to EA / ndarray" code anywhere?

Reply:
Not that I'm aware of.  I think it would be a useful pattern, wouldn't need to be limited to unary ops.

In the arithmetic ops stuff there are a bunch of places where code could be shared/simplified if the `_constructor` call below were somehow generalized to work for both `DataFrame` and `SparseDataFrame`.  I think that may be the sticking point here too.

Reply:
Well, hopefully SparseDataFrame can be deleted entirely :)

Reply:
> Well, hopefully SparseDataFrame can be deleted entirely :)

+1

Reply:
https://github.com/pandas-dev/pandas/issues/23087

will "resolve" this comment again